### PR TITLE
sdl2_mixer: updated to 2.8.0 and patch removed

### PR DESCRIPTION
**Pull request by NostalgicKitsune**
SDL2_mixer has been updated to 2.8.0, the latest available release, and uses `libxmp` by default instead of `libmodplug` (it remains as a fallback) for playing tracker modules.

"Fixed use-after-free in music_fluidsynth.c" patch has been removed because it has been fixed upstream.

https://github.com/FluidSynth/fluidsynth/issues/748
https://github.com/libsdl-org/SDL_mixer/commit/6160668079f91d57a5d7bf0b40ffdd843be70daf

### CODE REVIEW

**Comment by waddlesplash** [comment]
Why not just drop libmodplug?

File: media-libs/sdl2_mixer/sdl2_mixer-2.8.0.recipe [hunk 1]
```diff
@@ -9,16 +9,15 @@ COPYRIGHT="1997-2012 Sam Lantinga"
...
 SOURCE_DIR="SDL2_mixer-$portVersion"
-PATCHES="sdl2_mixer-$portVersion.patchset"
 
 ARCHITECTURES="all !x86_gcc2"
 SECONDARY_ARCHITECTURES="x86"
 
 PROVIDES="
 	sdl2_mixer$secondaryArchSuffix = $portVersion compat >= 2.0
-	lib:libSDL2_mixer_2.0$secondaryArchSuffix = 0.2.2 compat >= 0
+	lib:libSDL2_mixer_2.0$secondaryArchSuffix = 0.800.0 compat >= 0
```
Comment (waddlesplash) [inline media-libs/sdl2_mixer/sdl2_mixer-2.8.0.recipe:20]: Is this the real SOVERSION?

Comment (NostalgicKitsune) [reply media-libs/sdl2_mixer/sdl2_mixer-2.8.0.recipe:20]: Yes, the library when compiled reports the SOVERSION as 0.800.0

Comment (waddlesplash) [reply media-libs/sdl2_mixer/sdl2_mixer-2.8.0.recipe:20]: OK, very good.

**Comment by NostalgicKitsune** [comment]
Yes, I saw it only now, `libmodplug` is no longer needed since it uses `libxmp`.
I compiled without `libmodplug` and the compiler doesn't look for it, it only looks for `libxmp`

File: media-libs/sdl2_mixer/sdl2_mixer-2.8.0.recipe [hunk 2]
```diff
@@ -9,33 +9,32 @@ COPYRIGHT="1997-2012 Sam Lantinga"
 LICENSE="Zlib"
 REVISION="5"
```
Comment (waddlesplash) [inline media-libs/sdl2_mixer/sdl2_mixer-2.8.0.recipe:10]: This should be reduced to 1.

Comment (NostalgicKitsune) [reply media-libs/sdl2_mixer/sdl2_mixer-2.8.0.recipe:10]: Yeah, you're right

**Review by waddlesplash — APPROVED** [review]

**Pull request merged**